packages/x-charts/src/hooks/useColor.ts

```diff
@@ -0,0 +1,18 @@
+import * as React from 'react';
+import { ChartSeriesType } from '../internals';
+import { ColorContext } from '../context/ColorProvider';
+import { ColorProcessorsConfig } from '../models/plugin';
+
+export function useColorProcessor<T extends ChartSeriesType>(
+  seriesType: T,
+): ColorProcessorsConfig<ChartSeriesType>[T];
+export function useColorProcessor(): ColorProcessorsConfig<ChartSeriesType>;
+export function useColorProcessor(seriesType?: ChartSeriesType) {
```

Comment:
Following https://github.com/mui/mui-x/pull/13396

We could allow `ColorProcessor` to trim down the correct types itself
```tsx
// plugin.ts
export type ColorProcessor<T extends ChartSeriesType> = T extends ChartSeriesType
  ? (
      series: DefaultizedSeriesType<T>,
      xAxis?: AxisDefaultized,
      yAxis?: AxisDefaultized,
      zAxis?: ZAxisDefaultized,
    ) => (dataIndex: number) => string
  : never;
```

And then on this file

```suggestion
export function useColorProcessor<T extends ChartSeriesType>(seriesType: T): ColorProcessor<T>;
export function useColorProcessor(): ColorProcessorsConfig<ChartSeriesType>;
export function useColorProcessor(seriesType?: ChartSeriesType) {
```